### PR TITLE
check value option

### PR DESCRIPTION
**Pull request by jeffreyguo**
Check value in a input box, default option or all options in a selectbox.

### CODE REVIEW

File: src/SilverStripe/BehatExtension/Context/BasicContext.php [hunk 1]
```diff
@@ -942,4 +942,68 @@ public function spin($lambda, $wait = 60) {
             $backtrace[1]['function']
         ));
     }
+
+	/**
+	 * Check the value set in an input textbox(visible)
+	 * @example: Then the value in the "username" field should be "John Smith"
+	 * @Then /^the value in the "(?P<field>(?:[^"]|\\")*)" field should be "(?P<value>(?:[^"]|\\")*)"$/
+	 */
+	public function theValueInFieldShouldBe($field, $value) {
```
Comment (chillu) [inline src/SilverStripe/BehatExtension/Context/BasicContext.php]: @jeffreyguo Does this duplicate `Then /^the "(?P<field>(?:[^"]|\\")*)" field should contain "(?P<value>(?:[^"]|\\")*)"$/`? If its different, please follow the existing sentence structure, and explain clearly in the phpdoc on why this variation is required.

Comment (jeffreyguo) [inline src/SilverStripe/BehatExtension/Context/BasicContext.php]: @chillu, they are different. The existing one is check the value matched partially. My new one is to check the value matched exactly. I will update it later. Thanks!

**Comment by jeffreyguo** [comment]
Updated and squashed to commit[9977335].

**Comment by chillu** [comment]
So that's Mink's built in implementation:

```
/**
     * Checks, that form field with specified id|name|label|value has specified value.
     *
     * @Then /^the "(?P<field>(?:[^"]|\\")*)" field should contain "(?P<value>(?:[^"]|\\")*)"$/
     */
    public function assertFieldContains($field, $value)
    {
        $field = $this->fixStepArgument($field);
        $value = $this->fixStepArgument($value);
        $this->assertSession()->fieldValueEquals($field, $value);
    }
```

Then on to `WebAssert->fieldValueEquals()`. 

```
public function fieldValueEquals($field, $value, TraversableElement $container = null)
{
    $node   = $this->fieldExists($field, $container);
    $actual = $node->getValue();
    $regex  = '/^'.preg_quote($value, '/').'$/ui';

    $message = sprintf('The field "%s" value is "%s", but "%s" expected.', $field, $actual, $value);

    $this->assert((bool) preg_match($regex, $actual), $message);
}
```

So based on the regex in there, it checks for the exact value (starts with `^` and ends with `$`) - as it should. I think introducing a slightly different behaviour with a slightly different step wording without mentioning this new exact matching behaviour in there is going to confuse every user of the SilverStripe Behat extension. 

For `/^the option "([^"]*)" in the selectbox "([^"]*)" should (not |)be selected$/` - how is this different from the built-in `When /^(?:|I )select "(?P<option>(?:[^"]|\\")*)" from "(?P<select>(?:[^"]|\\")*)"$/`?

For `/^I should (not |)see "([^"]*)" in the selectbox "([^"]*)"$/`, you might have a case - can you please double check that there's no built-in step which does this? If there isn't, can you please ensure the wording of the step stays as close as possible to existing steps defined in this extension and the Mink core? For example, you're introducing the 'selectbox' naming where the built-in step just treats it as a named field, with the verb "select".

**Comment by jeffreyguo** [comment]
Thanks for your check, it's duplicated.

**Pull request closed without merging**